### ISSUE TITLE
unable to rke up with external-aws. node lookup fails if user doesn't set both internal-address and hostname-override

### ISSUE THREAD

**Issue by slickwarren** [rancher/rke#3445]
**RKE version:**
v1.5.0-rc11

**Docker version: (`docker version`,`docker info` preferred)**

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

**cluster.yml file:**
using aws nodes, set the public IP address and the following in the yaml:
```
cloud_provider:
  name: "external-aws"
  useInstanceMetadataHostname: true
```

**Steps to Reproduce:**

**Results:**
the cluster fails to come up, stating that the first node is not able to be found in the logs 
`FATA[0402] [ "<ip>" not found]`

In an offline discussion with @kinarashah , this appears to be due to the following:
internal address should be set
hostname-override should be set to the aws hostname that works with external-aws

**Comment by kinarashah** [comment]
The root cause is that RKE fails to find the node if both internal address and hostname override are empty. External address is not available to RKE at this stage of provisioning because it's populated by aws cloud controller manager after it is deployed. There are 2 options: 
1. Set internal address https://github.com/kinarashah/rke/blob/a74c388690524b9190dc1449a1bb154fdeb77bdc/k8s/node.go#L65 
2. Set hostname meeting the node conventions for out of tree cloud provider 
https://github.com/kinarashah/rke/blob/a74c388690524b9190dc1449a1bb154fdeb77bdc/k8s/node.go#L56

**Comment by jiaqiluo** [comment]
The fix was validated in RKE v1.5.0-rc11 per comments in https://github.com/rancher/rancher/issues/43624
